src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java

```diff
@@ -0,0 +1,279 @@
+
+        trimStringFields(tutorialGroup);
+        if (!tutorialGroupFromDatabase.getTitle().equals(tutorialGroup.getTitle())) {
+            checkTitleIsUnique(tutorialGroup);
+        }
+        overrideValues(tutorialGroup, tutorialGroupFromDatabase);
+
+        var updatedTutorialGroup = tutorialGroupRepository.save(tutorialGroupFromDatabase);
+        return ResponseEntity.ok(updatedTutorialGroup);
+    }
+
+    /**
+     * DELETE tutorial-groups/:tutorialGroupId/deregister/:studentLogin : deregister a student from a tutorial group.
+     *
+     * @param tutorialGroupId the id of the tutorial group
+     * @param studentLogin    the login of the student to deregister
+     * @return the ResponseEntity with status 204 (NO_CONTENT)
+     */
+    @DeleteMapping("/tutorial-groups/{tutorialGroupId}/deregister/{studentLogin:" + Constants.LOGIN_REGEX + "}")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<Void> deregisterStudent(@PathVariable Long tutorialGroupId, @PathVariable String studentLogin) {
+        log.debug("REST request to deregister {} student from tutorial group : {}", studentLogin, tutorialGroupId);
+        var tutorialGroupFromDatabase = this.tutorialGroupRepository.findByIdElseThrow(tutorialGroupId);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroupFromDatabase.getCourse(), null);
+
+        Optional<User> studentToDeregister = userRepository.findOneWithGroupsAndAuthoritiesByLogin(studentLogin);
+        if (studentToDeregister.isEmpty()) {
+            throw new EntityNotFoundException("User", studentLogin);
+        }
+        tutorialGroupService.deregisterStudent(studentToDeregister.get(), tutorialGroupFromDatabase);
+        return ResponseEntity.noContent().build();
+    }
+
+    /**
+     * POST tutorial-groups/:tutorialGroupId/register/:studentLogin : register a student to a tutorial group.
+     *
+     * @param tutorialGroupId the id of the tutorial group
+     * @param studentLogin    the login of the student to register
+     * @return the ResponseEntity with status 204 (NO_CONTENT)
+     */
+    @PostMapping("/tutorial-groups/{tutorialGroupId}/register/{studentLogin:" + Constants.LOGIN_REGEX + "}")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<Void> registerStudent(@PathVariable Long tutorialGroupId, @PathVariable String studentLogin) {
+        log.debug("REST request to register {} student to tutorial group : {}", studentLogin, tutorialGroupId);
+        var tutorialGroupFromDatabase = this.tutorialGroupRepository.findByIdElseThrow(tutorialGroupId);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroupFromDatabase.getCourse(), null);
+
+        Optional<User> userToRegister = userRepository.findOneWithGroupsAndAuthoritiesByLogin(studentLogin);
```

Comment:
Same here.

Reply:
Good idea! Done in 282a84a2a83ea0d17d0d693e02b3114c978652d5